internal/controllers/machine/machine_controller_phases.go

```diff
@@ -188,57 +139,57 @@ func (r *Reconciler) reconcileBootstrap(ctx context.Context, s *scope) (ctrl.Res
 	}
 
 	// Call generic external reconciler if we have an external reference.
-	externalResult, err := r.reconcileExternal(ctx, cluster, m, m.Spec.Bootstrap.ConfigRef)
+	obj, err := r.reconcileExternal(ctx, cluster, m, m.Spec.Bootstrap.ConfigRef)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			if !s.machine.DeletionTimestamp.IsZero() {
+				// Tolerate bootstrap object not found when the machine is being deleted.
+				// TODO: we can also relax this and tolerate the absence of the bootstrap ref way before, e.g. after node ref is set
+				return ctrl.Result{}, nil
+			}
+			log.Info("could not find bootstrap config object, requeuing", m.Spec.Bootstrap.ConfigRef.Kind, klog.KRef(m.Spec.Bootstrap.ConfigRef.Namespace, m.Spec.Bootstrap.ConfigRef.Name))
```

Comment:
```suggestion
			log.Info("Could not find bootstrap config object, requeuing", m.Spec.Bootstrap.ConfigRef.Kind, klog.KRef(m.Spec.Bootstrap.ConfigRef.Namespace, m.Spec.Bootstrap.ConfigRef.Name))
```
nit, probably same for infra